src/tracks/MidiClip.cpp

```diff
@@ -580,19 +580,24 @@ void MidiClip::cloneSteps()
 {
 	int oldLength = m_steps;
 	m_steps *= 2; // cloning doubles the track
-	for(int i = 0; i < oldLength; ++i )
-	{
-		Note *toCopy = noteAtStep( i );
-		if( toCopy )
-		{
-			setStep( oldLength + i, true );
-			Note *newNote = noteAtStep( oldLength + i );
-			newNote->setKey( toCopy->key() );
-			newNote->setLength( toCopy->length() );
-			newNote->setPanning( toCopy->getPanning() );
-			newNote->setVolume( toCopy->getVolume() );
-		}
+
+	const TimePos endPosition = TimePos::stepPosition(oldLength);
+
+	// Must reserve so that we don't accidentally grow the vector.
+	m_notes.reserve(m_notes.size() * 2);
+
+	instrumentTrack()->lock();
+
+	for (auto& note: m_notes)  {
```

Comment:
> Colons MUST be surrounded by spaces in range-based for loops and in base class specifications.

[Coding conventions: colons](https://github.com/LMMS/lmms/wiki/Coding-Conventions#colons)

> 4. Block braces SHOULD be on their own line

[Coding conventions: braces](https://github.com/LMMS/lmms/wiki/Coding-Conventions#braces)

`note` isn't mutated, so the reference ought to be declared `const`.

```suggestion
	for (const auto& note : m_notes)
	{
```